File: usql-core/src/main/java/com/usql/backend/GenerateOptions.java
package com.usql.backend;

/**
 * Options controlling SQL generation output.
 */
public record GenerateOptions(
    /**
     * Identifier quoting style.
     * Currently unused — all backends use their default quoting.
     * Reserved for future use when per-statement quoting control is needed.
     */
    QuoteStyle quoteStyle,

    /** Whether to add newlines/indentation */
    boolean prettyPrint,

    /** Indent string when prettyPrint is true */
    String indent,

    /** Emit source comments for traceability */
    boolean emitComments
) {
    public enum QuoteStyle {
        /** Use the dialect's default quoting rule */
        DEFAULT,
        /** Always quote all identifiers */
        ALWAYS,
        /** Only quote reserved words */
        RESERVED_ONLY,
        /** Never quote — caller takes responsibility */
        NEVER
    }

    public static final GenerateOptions DEFAULTS = new GenerateOptions(
        QuoteStyle.DEFAULT, true, "  ", false
    );

    public static final GenerateOptions MINIMAL = new GenerateOptions(
        QuoteStyle.DEFAULT, false, "", false
    );
}

File: usql-python/src/usql/backend/generate_options.py
"""Options controlling SQL generation output."""
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum, auto


class QuoteStyle(Enum):
    """Identifier quoting style."""
    DEFAULT = auto()       # Use the dialect's default quoting rule
    ALWAYS = auto()        # Always quote all identifiers
    RESERVED_ONLY = auto() # Only quote reserved words
    NEVER = auto()         # Never quote — caller takes responsibility


@dataclass(frozen=True)
class GenerateOptions:
    """Options controlling SQL generation output."""

    quote_style: QuoteStyle = QuoteStyle.DEFAULT
    pretty_print: bool = True
    indent: str = "  "
    emit_comments: bool = False

    DEFAULTS: GenerateOptions = None  # type: ignore
    MINIMAL: GenerateOptions = None  # type: ignore


# Class-level constants
GenerateOptions.DEFAULTS = GenerateOptions(QuoteStyle.DEFAULT, True, "  ", False)
GenerateOptions.MINIMAL = GenerateOptions(QuoteStyle.DEFAULT, False, "", False)
